app/jobs/update_all_loans_job.rb

```diff
@@ -0,0 +1,39 @@
+class UpdateAllLoansJob < TaskJob
+  def perform(_job_params)
+    task = task_for_job(self)
+    errors_by_loan = []
+    loans = Loan.all
+    updater = Accounting::Quickbooks::Updater.new
+    task.set_activity_message("syncing_with_quickbooks")
+    updater.qb_sync_for_loan_update
+    loans.each_with_index do |loan, index|
+      task.set_activity_message("updating_all_loans", {so_far: (index), total: loans.count})
+      begin
+        updater.update_loan(loan)
+      rescue StandardError => error
+        errors_by_loan << {loan_id: loan.id, message: error.message}
+        next
+      end
+    end
+    task.update(custom_error_data: errors_by_loan)
+    task.set_activity_message("finished_with_custom_error_data")
+  end
+
+  rescue_from(Accounting::Quickbooks::NotConnectedError) do |error|
```

Comment:
is there a better way to combine these two rescue_from overrides?

Reply:
Maybe, but unless we start adding a third/fourth error that we want to rescue from with very similar code, I'm not sure it's worth putting the energy into figuring it out, this is a very mild form of code duplication

Reply:
ok, thanks